### PR TITLE
#58 - mobile menu

### LINKED ISSUE #58: [Administrator panel] Mobile menu
Hamburger button should display menu.

image

### PR DESCRIPTION
**Pull request by AleksandraKozubal**
Added hamburger menu for smaller screens in admin's panel, moved run importers button to admin's panel.
Number of changed files is caused by changing a few routes, so the actual changes are not that extensive

It should close #58.

### CODE REVIEW

File: resources/js/Shared/Layout/AdminNavigation.vue [hunk 1]
```diff
@@ -18,48 +30,43 @@ const props = defineProps({
...
 
       <div class="mr-3.5 flex sm:hidden">
-        <button type="button" class="inline-flex items-center justify-center rounded-md p-2.5 text-gray-700" @click="mobileMenuOpen = true">
+        <button type="button" class="inline-flex items-center justify-center rounded-md p-2.5 text-gray-700" @click="toggleMobileMenu">
           <span class="sr-only">Open main menu</span>
           <Bars3Icon class="h-6 w-6" aria-hidden="true" />
         </button>
       </div>
+      <Dialog v-if="isMobileMenuOpened" as="div" class="z-30 lg:hidden" :open="isMobileMenuOpened" @close="toggleMobileMenu">
+        <div class="fixed inset-0 z-30 " />
```
Comment (EwelinaLasowy) [inline resources/js/Shared/Layout/AdminNavigation.vue:39]: Take care about unnecessary spaces.

File: resources/js/Shared/Layout/AdminNavigation.vue [hunk 2]
```diff
@@ -18,48 +29,43 @@ const props = defineProps({
...
+            </Link>
+            <button type="button" class="-m-2.5 rounded-md px-2.5 text-gray-700 sm:pt-4" @click="toggleMobileMenu">
+              <span class="sr-only">Close menu</span>
+              <XMarkIcon class="h-6 w-6" aria-hidden="true" />
+            </button>
+          </div>
+          <div class="mt-6 flow-root">
+            <div class="-my-6 divide-y divide-gray-500/10">
+              <div class="space-y-2 py-6">
+                <Link v-for="item in navigation" :key="item.name" :class="$page.url.startsWith(item.href)? 'bg-blumilk-50' : ''" :href="item.href" class="-mx-3 block rounded-lg px-3 py-2 text-base font-semibold leading-7 text-gray-800 hover:bg-blumilk-25" @click="clearFilters">
```
Comment (MichalMyskow) [inline resources/js/Shared/Layout/AdminNavigation.vue:52]: This **clearFilters** function does not exist in the **AdminNavigation** file and causes some console errors.
image

Comment (MichalMyskow) [inline resources/js/Shared/Layout/AdminNavigation.vue:52]:
```suggestion
                <Link v-for="item in navigation" :key="item.name" :class="{'bg-blumilk-50': $page.url.startsWith(item.href)}" :href="item.href" class="-mx-3 block rounded-lg px-3 py-2 text-base font-semibold leading-7 text-gray-800 hover:bg-blumilk-25" @click="clearFilters">
```
`:class` can be simplified

File: resources/js/Shared/Layout/AdminNavigation.vue [hunk 3]
```diff
@@ -18,48 +29,43 @@ const props = defineProps({
...
-            <span class="ml-3 hidden md:flex">Cities</span>
-          </li>
-        </Link>
-
-        <Link class="flex h-full md:h-fit" href="#">
-          <li
-            :class="props.url === '/admin/dashboard/statistics' ? 'bg-blumilk-50' : ''"
-            class="mx-auto flex w-11/12 items-center bg-blumilk-25 px-6 hover:bg-blumilk-50 md:rounded-lg md:px-2 md:py-3"
+        <Link v-for="item in navigation" :key="item.name" :href="item.href" class="flex h-full md:h-fit">
+          <div :class="$page.url.startsWith(item.href)? 'bg-blumilk-50' : ''"
```
Comment (MichalMyskow) [inline resources/js/Shared/Layout/AdminNavigation.vue:63]:
```suggestion
          <div :class="{'bg-blumilk-50': $page.url.startsWith(item.href)}"
```
Can be simplified

Comment (AleksandraKozubal) [reply resources/js/Shared/Layout/AdminNavigation.vue:63]: changed

File: resources/js/Shared/Layout/AdminNavigation.vue (hunk 2 above)
Comment (AleksandraKozubal) [reply resources/js/Shared/Layout/AdminNavigation.vue:52]: changed

File: resources/js/Shared/Layout/AdminNavigation.vue [hunk 4]
```diff
@@ -1,65 +1,84 @@
...
-            <MapPinIcon class="h-7 w-7" />
-            <span class="ml-3 hidden md:flex">Cities</span>
-          </li>
-        </Link>
-
-        <Link class="flex h-full md:h-fit" href="#">
-          <li
-            :class="props.url === '/admin/dashboard/statistics' ? 'bg-blumilk-50' : ''"
-            class="mx-auto flex w-11/12 items-center bg-blumilk-25 px-6 hover:bg-blumilk-50 md:rounded-lg md:px-2 md:py-3"
+        <Link v-for="item in navigation" :key="item.name" :href="item.href" class="flex h-full md:h-fit">
```
Comment (MichalMyskow) [inline resources/js/Shared/Layout/AdminNavigation.vue:75]: I would recommend changing all `<Link/>` components to `<InertiaLink/>` as defined in app.js.

File: resources/js/Shared/Layout/AdminNavigation.vue [hunk 5]
```diff
@@ -1,66 +1,85 @@
 <script setup>
-import { Link } from '@inertiajs/vue3'
-import { ChartBarIcon, ClipboardIcon, FlagIcon, MapPinIcon } from '@heroicons/vue/24/solid'
-import { Bars3Icon } from '@heroicons/vue/24/outline'
+import { Dialog, DialogPanel } from '@headlessui/vue'
+import { InertiaLink } from '@inertiajs/vue3'
```
Comment (MichalMyskow) [inline resources/js/Shared/Layout/AdminNavigation.vue:3]:
```suggestion
```
We don`t need this import now.

Also please change `<Link/>` in other views to `<InertiaLink/>` and remove **Link** import,

**Review by MichalMyskow — APPROVED** [review]

**Review by BarTracz — APPROVED** [review]

**Pull request merged**